tests/digraph/test_adjacency_matrix.py

```diff
@@ -85,6 +85,20 @@ def test_multigraph_sum_cast_weight_func(self):
         self.assertIsInstance(res, np.ndarray)
         self.assertTrue(np.array_equal(np.array([[0.0, 7.5], [0.0, 0.0]]), res))
 
+    def test_multigraph_sum_cast_weight_func_non_zero_null(self):
+        graph = retworkx.PyDiGraph()
+        node_a = graph.add_node("a")
+        node_b = graph.add_node("b")
+        graph.add_edge(node_a, node_b, 7.0)
+        graph.add_edge(node_a, node_b, 0.5)
```

Comment:
Why does this update the edge to `prev_weight + new_weight`? No multi-edges? Why not update length?

Reply:
This is adding a parallel edge between a and b, the test is specifically to test that for an adjacency matrix with parallel edges we sum the weights in the output matrix (which is the documented behavior).